components/layout/display_list_builder.rs

```diff
@@ -1931,12 +1938,19 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
         }
 
         state.current_scroll_root_id = original_scroll_root_id;
+        state.containing_block_scroll_root_id = original_containing_block_scroll_root;
         state.current_stacking_context_id = parent_stacking_context_id;
     }
 
     fn setup_scroll_root_for_block(&mut self, state: &mut DisplayListBuildState) -> ScrollRootId {
-        let containing_scroll_root_id = state.current_scroll_root_id;
+        // If this block is absolutely positioned, we should be clipped and positioned by
+        // the scroll root of our nearest ancestor that establishes a containing block.
+        let containing_scroll_root_id = match self.positioning() {
+            position::T::absolute => state.containing_block_scroll_root_id,
```

Comment:
@mrobinson do we need something similar for `position: fixed`? (keeping track of the root containing block?) Or is that handled somewhere else?

> If the element has 'position: fixed', the containing block is established by the viewport in the case of continuous media or the page area in the case of paged media.